tools/certify/keyio.cpp

```diff
@@ -0,0 +1,35 @@
+#include "keyio.hpp"
+#include <cryptopp/files.h>
+
+void save_key(const std::string& filename, const CryptoPP::BufferedTransformation& bt)
+{
+    CryptoPP::FileSink file(filename.c_str());
+
+    bt.CopyTo(file);
+    file.MessageEnd();
+}
+
+void load_key(const std::string& filename, CryptoPP::BufferedTransformation& bt)
+{
+    CryptoPP::FileSource file(filename.c_str(), true);
+
+    file.TransferTo(bt);
+    bt.MessageEnd();
+}
+
+vanetza::security::ecdsa256::KeyPair to_keypair(const vanetza::security::BackendCryptoPP::PrivateKey& private_key)
+{
+    vanetza::security::ecdsa256::KeyPair kp;
+
+    auto& private_exponent = private_key.GetPrivateExponent();
+    private_exponent.Encode(kp.private_key.key.data(), kp.private_key.key.size());
+
+    vanetza::security::BackendCryptoPP::PublicKey public_key;
+    private_key.MakePublicKey(public_key);
+
+    auto& public_element = public_key.GetPublicElement();
+    public_element.x.Encode(kp.public_key.x.data(), kp.public_key.x.size());
+    public_element.y.Encode(kp.public_key.y.data(), kp.public_key.y.size());
+
+    return kp;
+}
```

Comment:
@riebl Do you think it makes sense to make these functions from `BackendCryptoPP` public instead?

```cpp
/// create private key
PrivateKey generate_private_key();

/// derive public key from private key
PublicKey generate_public_key(const PrivateKey&);

/// adapt generic public key to internal structure
PublicKey internal_public_key(const ecdsa256::PublicKey&);

/// adapt generic private key to internal structure
PrivateKey internal_private_key(const ecdsa256::PrivateKey&);
```

These would also be usable in `genkey` then.

Reply:
@kelunik You seem to have found a good reason to make these methods public. At a first glance, these methods won't break any `BackendCryptoPP` invariant because `m_prng` is the only used member attribute.